File: java-web-spring-boot-3/src/main/java/com/vndevteam/javawebspringboot3/infrastructure/logging/RequestLoggingFilter.java
package com.vndevteam.javawebspringboot3.infrastructure.logging;

import jakarta.servlet.http.HttpServletRequest;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.lang.NonNull;
import org.springframework.web.filter.CommonsRequestLoggingFilter;

@Configuration
@Slf4j
public class RequestLoggingFilter extends CommonsRequestLoggingFilter {
    private static final String HEADER_NAME_API_KEY = "api-key";

    @Value("${app.logging.enable-log-request}")
    private boolean enableLogRequest;

    public RequestLoggingFilter() {
        setIncludeClientInfo(true);
        setIncludeQueryString(true);
        setIncludePayload(true);
        setMaxPayloadLength(10000);
        setIncludeHeaders(true);
        super.setBeforeMessagePrefix("REQUEST DATA: ");
        // Hide sensitive header if you need
        super.setHeaderPredicate(header -> !header.equals(HEADER_NAME_API_KEY));
    }

    @Override
    protected void beforeRequest(@NonNull HttpServletRequest request, @NonNull String message) {
        if (enableLogRequest) {
            log.info(message);
        }
    }

    @Override
    protected void afterRequest(@NonNull HttpServletRequest request, @NonNull String message) {
    }
}

File: kotlin-web-spring-boot-3/src/main/kotlin/com/vndevteam/kotlinwebspringboot3/infrastructure/logging/RequestLoggingFilter.kt
package com.vndevteam.kotlinwebspringboot3.infrastructure.logging

import jakarta.servlet.http.HttpServletRequest
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import org.springframework.web.filter.CommonsRequestLoggingFilter

@Configuration
class RequestLoggingFilter : CommonsRequestLoggingFilter() {
    companion object {
        private const val HEADER_NAME_API_KEY = "api-key"
    }

    @Value("\${app.logging.enable-log-request}") val enableLogRequest: Boolean = false

    init {
        isIncludeClientInfo = true
        isIncludeQueryString = true
        isIncludePayload = true
        maxPayloadLength = 10000
        isIncludeHeaders = true
        super.setBeforeMessagePrefix("REQUEST DATA: ")
        // Hide sensitive header if you need
        super.setHeaderPredicate { header -> header != HEADER_NAME_API_KEY }
    }

    override fun beforeRequest(request: HttpServletRequest, message: String) {
        if (enableLogRequest) {
            logger.info(message)
        }
    }

    override fun afterRequest(request: HttpServletRequest, message: String) {}
}
